File: basics/04-unique-characters/Solution.java
import java.util.*;

class Solution {
    public List<Character> findUnique(String s) {
        Map<Character, Integer> count = new HashMap<>();
        for (char c : s.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        List<Character> result = new ArrayList<>();
        for (Map.Entry<Character, Integer> e : count.entrySet()) {
            if (e.getValue() == 1) result.add(e.getKey());
        }
        return result;
    }
}

File: basics/04-unique-characters/Solution.cpp
#include <string>
#include <vector>
#include <unordered_map>

std::vector<char> findUnique(const std::string& s) {
    std::unordered_map<char, int> count;
    for (char c : s) count[c]++;
    std::vector<char> result;
    for (auto& [c, freq] : count) {
        if (freq == 1) result.push_back(c);
    }
    return result;
}
